tests/tests_unit/approval_client/client.py

```diff
@@ -412,12 +457,72 @@ def _create_retrieve_method(self, resource: APIResource, mock_method: str, clien
         resource_cls = resource.resource_cls
         read_list_cls = resource.list_cls
 
+        def _create_verification_resource(*args, **kwargs):
+            # Note that the written version of the resource does not contain the serves set variables,
+            # so we need to set these manually
+            if resource_cls is Space:
+                ids = list(*args)
+                spaces = [
+                    Space(
+                        space=space,
+                        is_global=False,
+                        last_updated_time=1,
+                        created_time=1,
+                    )
+                    for space in ids
+                ]
+                return read_list_cls(spaces, cognite_client=client)
+            elif resource_cls is DataSet:
+                if "external_ids" in kwargs:
+                    external_ids = kwargs["external_ids"]
+                elif "external_id" in kwargs:
+                    external_ids = [kwargs["external_id"]]
+                else:
+                    raise NotImplementedError("No external_ids or external_id in kwargs")
```

Comment:
Should this be? Same question for the two next occurrences as well
```suggestion
                    raise RuntimeError("No external_ids or external_id in kwargs")
```